### DISCUSSION TITLE
script does not appear to execute

### DISCUSSION THREAD

**Discussion by mldulaney** [ipxe/ipxe#1318] [Q&A] [answered]
so, i have a problem:
if i download http://boot.ipxe.org/undionly.kpxe and drop it in my tftp server, it works (infinite boot loop as advertised), and i want to chainload.
so, per:  https://ipxe.org/howto/chainloading
  #!ipxe
  echo bacon
  dhcp
  echo Loading next link in chain
  chain http://boot.home.lan/boot/boot.php
and then i build:
make bin/undionly.kpxe EMBED=demo.ipxe
and i put that on the tftp server and it gets as far as saying that it is ipxe but does not appear to run the chain load script

**Comment by NiKiZe** [comment, marked as answer]
Verify that the file has valid magic to be a iPXE script, that is that it actually starts with `#!ipxe` and not some other magic BOM bytes or similar before it.

You could also verify that the script itself works by pressing ctrl-b and manually chain into the script that you temporarily put on your TFTP as will.

You can also use `imgfetch` and then `imgstat` to verify that iPXE detects it as a valid script.

**Comment by mldulaney** [comment]
there was an extraneous byte
